runtime/vam/expr/arith.go

```diff
@@ -48,5 +51,90 @@ func (a *Arith) eval(vecs ...vector.Any) vector.Any {
 	if !ok {
 		return vector.NewStringError(a.zctx, coerce.ErrIncompatibleTypes.Error(), lhs.Len())
 	}
-	return f(lhs, rhs)
+	out, ok := f(lhs, rhs)
+	if !ok {
+		return a.evalDivideByZero(kind, lhs, rhs)
+	}
+	return out
+}
+
+func (a *Arith) evalDivideByZero(kind vector.Kind, lhs, rhs vector.Any) vector.Any {
+	if a.opCode != vector.ArithDiv && a.opCode != vector.ArithMod {
+		panic("invalid op for evalDivideByZero")
+	}
+	var errs []uint32
+	var out vector.Any
+	switch kind {
+	case vector.KindInt:
+		var ints []int64
+		nulls := vector.NewBoolEmpty(lhs.Len(), nil)
+		for i := range lhs.Len() {
+			l, lnull := vector.IntValue(lhs, i)
+			r, rnull := vector.IntValue(rhs, i)
+			if lnull || rnull {
+				nulls.Set(i)
+				ints = append(ints, 0)
+				continue
+			}
+			if r == 0 {
+				errs = append(errs, i)
+				continue
+			}
+			if a.opCode == vector.ArithDiv {
+				ints = append(ints, l/r)
+			} else {
+				ints = append(ints, l%r)
+			}
+		}
+		out = vector.NewInt(super.TypeInt64, ints, nulls)
+	case vector.KindUint:
+		var uints []uint64
+		nulls := vector.NewBoolEmpty(lhs.Len(), nil)
+		for i := range lhs.Len() {
+			l, lnull := vector.UintValue(lhs, i)
+			r, rnull := vector.UintValue(rhs, i)
+			if lnull || rnull {
+				nulls.Set(i)
+				uints = append(uints, 0)
+				continue
+			}
+			if r == 0 {
+				errs = append(errs, i)
+				continue
+			}
+			if a.opCode == vector.ArithDiv {
+				uints = append(uints, l/r)
+			} else {
+				uints = append(uints, l%r)
+			}
+		}
+		out = vector.NewUint(super.TypeUint64, uints, nulls)
+	case vector.KindFloat:
+		if a.opCode == vector.ArithMod {
+			panic("mod on float")
+		}
+		var floats []float64
+		nulls := vector.NewBoolEmpty(lhs.Len(), nil)
+		for i := range lhs.Len() {
+			l, lnull := vector.FloatValue(lhs, i)
+			r, rnull := vector.FloatValue(rhs, i)
+			if lnull || rnull {
+				nulls.Set(i)
+				floats = append(floats, 0)
+				continue
+			}
+			if r == 0 {
+				errs = append(errs, i)
+				continue
+			}
+			floats = append(floats, l/r)
+		}
+		out = vector.NewFloat(super.TypeFloat64, floats, nulls)
```

Comment:
Floating point division by zero isn't usually an error. (The IEEE 754 spec says it results in NaN, +Inf, or -Inf depending on the operands.)
```suggestion
```